### PR TITLE
Disable lightup logger if user adds AI logger

### PR DESCRIPTION
**Pull request by pakrym**
Also prevents a situation when default `IConfiguration` is not wired to `LogginFactory` and lightup logger is sending all events to AI.

### CODE REVIEW

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 1]
```diff
@@ -22,13 +22,30 @@ public void ConfigureServices(IServiceCollection services)
             services.AddMvc();
         }
 
-        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
+        public void ConfigureJavaScript(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
         {
-            var logger = loggerFactory.CreateLogger("Requests");
-
+            loggerFactory.AddConsole(LogLevel.Debug);
```
Comment (BrennanConroy) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: wrong

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 2]
```diff
@@ -22,13 +22,30 @@ public void ConfigureServices(IServiceCollection services)
...
+        public void ConfigureDefaultLogging(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
+        {
+            loggerFactory.AddConsole(LogLevel.Debug);
+            ConfigureLoggingMiddleware(app, loggerFactory);
+        }
+
+        public void ConfigureCustomLogging(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
+        {
+            loggerFactory.AddConsole(LogLevel.Debug);
+            loggerFactory.AddApplicationInsights(serviceProvider, (s, level) => s.Contains("o"));
```
Comment (BrennanConroy) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: Whats with the "o"?

File: src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsLoggerStartupFilter.cs [hunk 3]
```diff
@@ -11,15 +11,35 @@ namespace Microsoft.AspNetCore.ApplicationInsights.HostingStartup
...
+
                 // We need to disable filtering on logger, filtering would be done by LoggerFactory
-                loggerFactory.AddApplicationInsights(builder.ApplicationServices, _noFilter);
+                var loggerEnabled = true;
+                Action disableCallback = () => loggerEnabled = false;
+
+                if (loggerFactory is LoggerFactory stronglyTypedLoggerFactory &&
+                    ApplicationInsightsLoggerConfiguration.HasLoggingConfigured(stronglyTypedLoggerFactory.Configuration))
+                {
+                    // We detected that logger settings got to LoggerFactory confi
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsLoggerStartupFilter.cs]: finish the comment

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 4]
```diff
@@ -110,8 +84,11 @@ public static void Main(string[] args)
             var host = new WebHostBuilder()
                 .ConfigureLogging((hostingContext, factory) =>
                 {
-                    factory.UseConfiguration(hostingContext.Configuration.GetSection("Logging"))
-                           .AddConsole();
+                    if (config["WIRE_LOGGING_CONFIGURATION"]?.ToLowerInvariant() != "false")
```
Comment (JunTaoLuo) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: use the iconfiguration on the hostingContext.

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs (hunk 2 above)
Comment (pakrym) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: Just any random filter that is subset of default and `settings.json`

File: sample/ApplicationInsightsHostingStartupSample/Startup.cs [hunk 5]
```diff
@@ -22,13 +22,27 @@ public void ConfigureServices(IServiceCollection services)
             services.AddMvc();
         }
 
-        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
+        public void ConfigureJavaScript(IApplicationBuilder app, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
```
Comment (BrennanConroy) [inline sample/ApplicationInsightsHostingStartupSample/Startup.cs]: Remove `ILoggerFactory` from these methods

File: test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/ApplicationInsightsLoggingTest.cs [hunk 6]
```diff
@@ -41,6 +154,9 @@ public ApplicationInsightsLoggingTest(ITestOutputHelper output) : base(output)
                         new KeyValuePair<string, string>(
                             "HOME",
                             Path.Combine(GetApplicationPath(), "home")),
+                        new KeyValuePair<string, string>(
+                            "ASPNETCORE_WIRE_LOGGING_CONFIGURATION",
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/ApplicationInsightsLoggingTest.cs]: This is different from the one used in the sample

Comment (pakrym) [inline test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/ApplicationInsightsLoggingTest.cs]: It has a prefix that is stripped by configuration.

**Comment by pakrym** [comment]
@Eilon approval to merge this?  https://github.com/aspnet/AzureIntegration/issues/26

File: src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsLoggerConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        public static void ConfigureLogging(IConfigurationBuilder configurationBuilder)
+        {
+            // Skip adding default rules when debugger is attached
+            // we want to send all events to VS
+            if (!Debugger.IsAttached)
+            {
+                configurationBuilder.AddInMemoryCollection(GetDefaultLoggingSettings());
+            }
+
+            var home = Environment.GetEnvironmentVariable("HOME");
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsLoggerConfiguration.cs]: Antares is the main scenario for this feature and we know exactly where this file is supposed to be.

File: src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsLoggerConfiguration.cs [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            new KeyValuePair<string, LogLevel>("Microsoft", LogLevel.Warning),
+            new KeyValuePair<string, LogLevel>("System", LogLevel.Warning),
+            new KeyValuePair<string, LogLevel>(null, LogLevel.Information)
+        };
+
+        public static void ConfigureLogging(IConfigurationBuilder configurationBuilder)
+        {
+            // Skip adding default rules when debugger is attached
+            // we want to send all events to VS
+            if (!Debugger.IsAttached)
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsLoggerConfiguration.cs]: Unfortunately this is the only way to do it for preview1, we are going to review filter configuration code after making changes to logging in preview2.

**Pull request merged**